hooks/shotgun_fields.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that controls how Shotgun data should be displayed in the info panel app. 
+"""
+import sgtk
+import os
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+class ShotgunConfiguration(HookBaseClass):
+        
+    def get_list_item_definition(self, entity_type):
+        """
+        Controls the rendering of items in the listings.
+        """
+        
+        # define a set of defaults
+        values = {
+            "top_left": "<big>{code}</big>",
+            "top_right": "{updated_at}",
+            "body": "<b>By:</b> {created_by}{[<br><b>Description:</b> ]description}"            
+            } 
+        
+        # override 
+        if entity_type == "PublishedFile":
+            
+            values["top_left"] = "<big>{name} v{version_number}</big>"
+            values["top_right"] = "{created_at}"
+            values["body"] = """
+                {published_file_type} by {created_by}<br>
+                <b>Comments:</b> {description}
+                """            
+    
+        elif entity_type == "Note":
+            
+            values["top_left"] = "<big>{created_by}</big>"
+            values["body"] = "{content}"     
+    
+        elif entity_type == "Version":
+            
+            values["body"] = """
+                <b>By:</b> {user|created_by}<br>
+                <b>Comments:</b> {description}
+                """
+        
+        elif entity_type == "Task":
+            
+            values["top_left"] = "<big>{content}</big>"
+            values["body"] = """
+                Assigned to {task_assignees}<br>
+                Status: {sg_status_list}
+                {[<br>Start: ]start_date}
+                {[<br>Due: ]due_date}
+                """            
+
+        return values
+        
+    
+    def get_all_fields(self, entity_type):
+        """
+        Define which fields should be displayed in the info tab
+        for a given entity
+        """
```

Comment:
See previous comment - i was thinking about all of the fields driven by the field templating strings as a unit, so seemed to make sense to keep them all together in a hook.